logagg/forward/forwarder.py

```diff
@@ -104,6 +125,70 @@ def read_from_q(self):
             except pymongo.errors.ServerSelectionTimeoutError:
                 self.log.exception('Push to mongo and ack to nsq failed')
 
+    def parse_msg_to_send_influxdb(self, msgs_list):
+        series = []
+        for msg in msgs_list:
+            time = msg.get('timestamp')
+            if msg.get('type') == self.METRIC:
+                event = msg.get('data').get('event')
+                metric = event.get('req_fn')
+                fields = event
+                tags = {}
+                if metric == self.API:
+                    tags['name'] = event.get('name')
+                    tags['host'] = event.get('host')
+                    tags['fn'] = event.get('fn')
+                    tags['success'] = event.get('success')
+                else:
+                    tags['host'] = event.get('host')
+                    tags['name'] = event.get('name')
+
+                pointValues = {
+                    "time": time,
+                    "measurement": metric,
+                    "fields": fields,
+                    "tags": tags
+                   }
+                series.append(pointValues)
+
+            elif msg.get('handler') == self.NGINX_HANDLER:
+                event = msg.get('data')
+                request = event.get('request')
+                measurement = self.NGINX_METRIC
+                host =  msg.get('host')
+                pointValues = {
+                    "time": time,
+                    "measurement": measurement,
+                    "fields": event,
+                    "tags": {
+                    "host": host,
+                    "request": request
+                    }
+                }
+                series.append(pointValues)
+
+            elif msg.get('handler') == self.DJANGO_HANDLER:
+                if isinstance(msg.get('data'), dict) and isinstance(msg.get('data').get('message'), dict):
+                    event = msg.get('data').get('message')
+                    if 'processing_time' in event:
+                        host = event.get('host')
+                        url = event.get('url')
+                        user = event.get('user')
+                        measurement = self.DJANGO_METRIC
+                        pointValues = {
+                            "time": time,
+                            "measurement": measurement,
+                            "fields": event,
+                            "tags": {
+                                "host": host,
+                                "url": url,
+                                "user": user
+                                }
+                            }
+                        series.append(pointValues)
+
+        return series
```

Comment:
@AnushaKonduru This function has become too long that it can't be accessible.
Can you please write a sub-functions to move some functionality there?

Reply:
Sure @venugopalkandru, will do.